homeassistant/components/flipr/__init__.py

```diff
@@ -28,6 +48,29 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
     if unload_ok:
-        hass.data[DOMAIN].pop(entry.entry_id)
+        hass.data[DOMAIN].pop(CONF_ENTRY_FLIPR_COORDINATORS)
 
     return unload_ok
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate config entry."""
+    _LOGGER.info("Migration of flipr config from version %s", entry.version)
+
+    if entry.version == 1:
+        # In version 1, we have flipr id as config entry unique id and one device per config entry.
+        # We need to migrate to a new config entry that may contain multiple devices. So we change the title and entry data to match config_flow evolution.
+        login = entry.data[CONF_EMAIL]
+        new_title = f"Flipr {login}"
+
+        new_data = {**entry.data}
+        # We do not store anymore the flipr_id in the config entry.
+        new_data.pop("flipr_id")
+
+        hass.config_entries.async_update_entry(
+            entry, data=new_data, title=new_title, version=2
+        )
+
+        _LOGGER.info("Migration of flipr config to version 2 successful")
+
+    return True
```

Comment:
I don't even think we have to do a migration after all.
- We shouldn't touch the title, as that is userland
- There is no direct need to pop the flipr id, allowing the user to rollback.

We do have to think about what if people have the integration setup twice, what do we do then?

Reply:
Title change was just to be aligned with title positionned by the config_flow v1. But Ok, the user can have changed it.
Yes the the pop of flipr_id was just to not leave useless information in config entry.
For people having the integration setup twice, which is very rare, I will implement a detection of if it and notify user with an error.
So globaly I remove the migration and keep it in v1.